pkg/cache/cache_test.go

```diff
@@ -2553,3 +2623,167 @@ func messageOrEmpty(err error) string {
 	}
 	return err.Error()
 }
+
+func TestClusterQueuesUsingAdmissionChecks(t *testing.T) {
+	admissionCheck1 := utiltesting.MakeAdmissionCheck("ac1").Obj()
+	admissionCheck2 := utiltesting.MakeAdmissionCheck("ac2").Obj()
+	fooCq := utiltesting.MakeClusterQueue("fooCq").
+		AdmissionChecks(admissionCheck1.Name).
+		Obj()
+	barCq := utiltesting.MakeClusterQueue("barCq").Obj()
+	fizzCq := utiltesting.MakeClusterQueue("fizzCq").
+		AdmissionChecks(admissionCheck1.Name, admissionCheck2.Name).
+		Obj()
+
+	cases := map[string]struct {
+		clusterQueues              []*kueue.ClusterQueue
+		wantInUseClusterQueueNames []string
+	}{
+		"single clusterQueue with check in use": {
+			clusterQueues: []*kueue.ClusterQueue{
+				fooCq,
+			},
+			wantInUseClusterQueueNames: []string{fooCq.Name},
+		},
+		"single clusterQueue with no checks": {
+			clusterQueues: []*kueue.ClusterQueue{
+				barCq,
+			},
+		},
+		"multiple clusterQueues with checks in use": {
+			clusterQueues: []*kueue.ClusterQueue{
+				fooCq,
+				barCq,
+				fizzCq,
+			},
+			wantInUseClusterQueueNames: []string{fooCq.Name, fizzCq.Name},
+		},
+	}
+	for name, tc := range cases {
+		t.Run(name, func(t *testing.T) {
+			cache := New(utiltesting.NewFakeClient())
+			cache.AddOrUpdateAdmissionCheck(admissionCheck1)
+			cache.AddOrUpdateAdmissionCheck(admissionCheck2)
+
+			for _, cq := range tc.clusterQueues {
+				if err := cache.AddClusterQueue(context.Background(), cq); err != nil {
+					t.Errorf("failed to add clusterQueue %s", cq.Name)
+				}
+			}
+
+			cqs := cache.ClusterQueuesUsingAdmissionCheck(admissionCheck1.Name)
+			if diff := cmp.Diff(tc.wantInUseClusterQueueNames, cqs, cmpopts.SortSlices(func(a, b string) bool {
+				return a < b
+			})); len(diff) != 0 {
+				t.Errorf("Unexpected flavor is in use by clusterQueues (-want,+got):\n%s", diff)
+			}
+		})
+	}
+}
+
+func TestClusterQueueRediness(t *testing.T) {
+
+	baseFalvor := utiltesting.MakeResourceFlavor("flavor1").Obj()
+	baseCheck := utiltesting.MakeAdmissionCheck("check1").Obj()
+	baseQueue := utiltesting.MakeClusterQueue("queue1").
+		ResourceGroup(
+			*utiltesting.MakeFlavorQuotas(baseFalvor.Name).
+				Resource(corev1.ResourceCPU, "10", "10").Obj()).
+		AdmissionChecks(baseCheck.Name).
+		Obj()
+
+	cases := map[string]struct {
+		queues      []*kueue.ClusterQueue
+		flavors     []*kueue.ResourceFlavor
+		checks      []*kueue.AdmissionCheck
+		queueName   string
+		terminate   bool
+		wantStatus  metav1.ConditionStatus
+		wantReason  string
+		wantMessage string
```

Comment:
Can you also add `wantTerminating` and `wantActive` to test the values of the ClusterQueueTerminating and ClusterQueueActive functions?

That way, we can ensure through unit tests that the implementations don't deviate.